core/src/main/clojure/xtdb/vector/writer.clj

```diff
@@ -604,9 +619,16 @@
     (dotimes [n child-count]
       (let [src-type-id (or (when type-ids (aget type-ids n))
                             n)
-            col-type (types/field->col-type (.get child-fields n))]
-        (aset copier-mapping src-type-id (.rowCopier (.writerForType dest-col col-type)
-                                                     (.getVectorByType src-vec src-type-id)))))
+            ^Field child-field (.get child-fields n)
+            child-field-name (.getName child-field)
+            col-type (types/field->col-type child-field)]
+        (aset copier-mapping src-type-id
+              ;; HACK to make things work for named duv legs
+              (if-not (str/starts-with? child-field-name (types/col-type->field-name col-type))
```

Comment:
there's a col-type->duv-leg(?) function that we likely could/should use here instead of the prefix of the col-name

Reply:
As discussed: removed the field-name indexing and did the comparison of field-name and `(col-type->field-name)`.